### PR TITLE
Add make_pub_crate code action to ra_editor

### PR DESCRIPTION
**Pull request by h-michael**
implement #170

### CODE REVIEW

**Comment by DJMcNab** [comment]
Does this handle items which are already `pub` in some capacity gracefully. Looking at the code, I'd guess not

File: crates/ra_editor/src/code_actions.rs [hunk 1]
```diff
@@ -333,4 +354,30 @@ fn foo() {
         );
     }
 
+    #[test]
+    fn test_make_pub_crate() {
+        check_action(
+            "<|>fn foo() {}",
+            "<|>pub (crate) fn foo() {}",
```
Comment (matklad) [inline crates/ra_editor/src/code_actions.rs]: The space after `pub` should be removed per usual styleguide.

**Comment by matklad** [comment]
Yep, I think ideally we should not make this intention available if there's already an existing visibility. To implement this, you'll need to get the `FnDef`, `StructDef` etc ancsetor of keyword and check if it has visibility.

**Comment by h-michael** [comment]
@DJMcNab @matklad
Thanks for your reviewing.
I'll fix it in this week.

**Comment by DJMcNab** [comment]
How does this handle e.g `unsafe fn foo()`? I would say rather than using previous sibling, you should get the parent declaration then see if it has a `Visibility`

**Comment by h-michael** [comment]
@DJMcNab Thank you for pointing out my overlooking.
I've changed to check parent declaration. CC: @matklad

**Comment by matklad** [comment]
LGTM, thanks @h-michael !

bors r+

**Pull request merged**